lib/phoenix_live_view.ex

```diff
@@ -681,6 +807,129 @@ defmodule Phoenix.LiveView do
   """
   defdelegate push_event(socket, event, payload), to: Phoenix.LiveView.Utils
 
+  @doc """
+  Allows an upload for the provided name.
+
+  ## Options
+
+    * `:accept` - Required. A list of unique file type specifiers or the
+      atom :any to allow any kind of file. For example, `[".jpeg"]`, `:any`, etc.
+
+    * `:max_entries` - The maximum number of selected files to allow per
+      file input. Defaults to 1.
+
+    * `:max_file_size` - The maximum file size in bytes to allow to be uploaded.
+      Defaults 8MB. For example, `12_000_000`.
+
+    * `:chunk_size` - The chunk size in bytes to send when uploading.
+      Defaults `64_000`.
+
+    * `:chunk_timeout` - The time in milliseconds to wait before closing the
+      upload channel when a new chunk has not been received. Defaults `10_000`.
+
+    * `:external` - The 2-arity function for generating metadata for external
+      client uploaders. See the Uploads section for example usage.
+
+  Raises when a previously allowed upload under the same name is still active.
+
+  ## Examples
+
+      allow_upload(socket, :avatar, accept: ~w(.jpg .jpeg), max_entries: 2)
+      allow_upload(socket, :avatar, accept: :any)
+  """
+  defdelegate allow_upload(socket, name, options), to: Phoenix.LiveView.Upload
+
+  @doc """
+  Revokes a previously allowed upload from `allow_upload/3`.
+
+  ## Examples
+
+      disallow_upload(socket, :avatar)
+  """
+  defdelegate disallow_upload(socket, name), to: Phoenix.LiveView.Upload
+
+  @doc """
+  Cancels an upload for the given entry.
+
+  ## Examples
+
+      <%= for entry <- @uploads.avatar.entries do %>
+        ...
+        <button phx-click="cancel-upload" phx-value-ref="<%= entry.ref %>">cancel</button>
+      <% end %>
+
+      def handle_event("cancel-upload", %{"ref" => ref}, socket) do
+        {:noreply, cancel_upload(socket, :avatar, ref)}
+      end
+  """
+  defdelegate cancel_upload(socket, name, entry_ref), to: Phoenix.LiveView.Upload
+
+  @doc """
+  Returns the completed and in progress entries for the upload.
+
+  ## Examples
+
+      case uploaded_entries(socket, :photos) do
+        {[_ | _] = completed, []} ->
+          # all entries are completed
+
+        {[], [_ | _] = in_progress} ->
+          # all entries are still in progress
+      end
+  """
+  defdelegate uploaded_entries(socket, name), to: Phoenix.LiveView.Upload
+
+  @doc ~S"""
+  Consumes the uploaded entries.
+
+  Raises when there are still entries in progress.
```

Comment:
Is it guaranteed that before the form submits, all entries will be completed? If so, we should probably mention it. :)

Reply:
yes, guaranteed. Good point!